testsuite/tests/kuadrant/gateway/conftest.py

```diff
@@ -0,0 +1,34 @@
+"""Conftest for gateway tests"""
+import pytest
+
+from testsuite.httpx.auth import HttpxOidcClientAuth
+from testsuite.policy.authorization.auth_policy import AuthPolicy
+
+
+@pytest.fixture(scope="module")
+def kuadrant(kuadrant):
+    """Skip if not running on Kuadrant"""
+    if not kuadrant:
+        pytest.skip("Gateway tests are only for Kuadrant")
+    return kuadrant
+
+
+@pytest.fixture(scope="module", autouse=True)
+def gateway_wait_for_ready(gateway):
+    """Waits for gateway to be ready"""
+    gateway.wait_for_ready()
+
+
+@pytest.fixture(scope="module")
+def authorization(gateway_wait_for_ready, gateway, route, oidc_provider, authorization_name, openshift, module_label):
```

Comment:
I do see the reason to explicitly request `gateway_wait_for_ready` fixture but at the same time I don't see why the fixture is autouse at the same time.
Another way how to refactor it would be to create a `ready_gateway` fixture returning a Gateway on which the `wait_for_ready()` was applied and then request `ready_gateway` instead of `gateway_wait_for_ready` and `gateway` simultaneously.

Reply:
Good suggestion :+1: